[Nathanael M. Tan] iP

itsNatTan opened the pull request:
# EpochMind
>“Your mind is for having ideas, not holding them.” – David Allen

EpochMind frees your mind from having to remember things you need to do. It's,
- Text-based
- Easy to learn
- ~~FAST~~ _SUPER_ FAST to use

All you need to do is,
1. Download it from [here](https://github.com/itsNatTan/ip).
2. Double-click it.
3. Add your tasks.
4. Let it manage your tasks for you 😉 

And it is **FREE**!

Features:
- [x] Managing tasks
- [ ] Managing deadlines (coming soon)
- [ ] Reminders (coming soon)


If you Java programmer, you can use it to practice Java too. Here's the main method:

``` Java
public class Main {
    public static void main(String[] args) {
        Application.launch(EpochMind.class, args);
    }
}
```

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

itsNatTan committed:
Renamed to EpochMind Added greeting and exit messages

itsNatTan committed:
Echoes the command and exits when command is bye

itsNatTan committed:
Create list functionality

itsNatTan committed:
Created Task class
Created mark and unmark functionality
Error handling

itsNatTan committed:
Created subclasses of Task
Error Handling

itsNatTan committed:
Exception class created
Handling more errors

itsNatTan committed:
Add delete function

itsNatTan committed:
Error handling of invalid arguments

itsNatTan committed:
Edited the bat and sh file for automated testing

itsNatTan committed:
Add save functionality
Add javadocs

itsNatTan committed:
Add DateTime and some extra functionalities

itsNatTan committed:
Add new classes for OOP (Parser, Storage, TaskList, Ui)
Add save method that will save at specified location

itsNatTan committed:
Put classes into appropriate packages

itsNatTan committed:
Merge branch 'add-gradle-support'

itsNatTan committed:
Made JUnit tests
Made changes to list, dueby and overdue methods

itsNatTan committed:
Jar

itsNatTan committed:
Add JavaDocs

itsNatTan committed:
Edit to follow coding standards

itsNatTan committed:
Add find functionality

itsNatTan committed:
Merge pull request #1 from itsNatTan/branch-Level-9

Add find functionality

itsNatTan committed:
Merge pull request #2 from itsNatTan/branch-A-CodingStandard

Edit to follow coding standards

itsNatTan committed:
Merge pull request #3 from itsNatTan/branch-A-JavaDoc

Add JavaDocs

itsNatTan committed:
More docs

Gabriel4357 commented on src/main/java/util/Ui.java:
```diff
@@ -0,0 +1,52 @@
+    /**
+     * Print horizontal line breaks
+     */
+    public static void line() {
```
Perhaps a verb could be added in front of the method line() since it is a method. Maybe displayLine() or printLine() ? There are other areas where this is seen as well such as greeting() or just bye(). 

``` suggestion 
    public static void displayLine() {
```

WZWren replied:
This seems like a solid and sensible change.

Gabriel4357 commented on src/main/java/util/Ui.java:
```diff
@@ -0,0 +1,52 @@
+            "`----'      `---`               `----'             '---'            ---`-'                `----'    \n" +
+            "                                                                                                    \n";
+
+    public static String[] exitMessage = {"May you seek the truth.",
```
Since this is an array, maybe it could be renamed to exitMessages (as a plural) 
```suggestion
    public static String[] exitMessages = {"May you seek the truth.",
 
```

Gabriel4357 commented on src/main/java/util/Parser.java:
```diff
@@ -0,0 +1,185 @@
+                System.out.println(e);
+                break;
+        }
+    }
```
There should be no indentation for case clauses.

Gabriel4357 commented on src/main/java/util/Parser.java:
```diff
@@ -0,0 +1,185 @@
+    }
+
+    /**
+     * Add an event task
```
Javadocs should use Adds rather than Add. This can be seen in other parts of the code too.

Gabriel4357 commented on src/main/java/util/Parser.java:
```diff
@@ -0,0 +1,185 @@
+            EpochMindException e = new EpochMindException("Thou hast not specified a task");
+            System.out.println(e);
+        } else {
+            tasks.todo(restOfCommand);
```
NIce. Even though the else block contains of one statement, you still put Egyptian style of braces.

Gabriel4357 commented on src/main/java/task/TaskList.java:
```diff
@@ -0,0 +1,247 @@
+    /**
+     * Find all tasks that contains the string searchBy
+     * @param searchBy The string to searchBy
+     * @return String representation of all the tasks that contains the string
```
Should have a spacing between method header and @param

Gabriel4357 reviewed (commented):
Overall great code 👍 Only a small number of minor coding practice violations. 
Keep up the good work! 😄

itsNatTan committed:
Add checkstyle functionality
Fix some of code style errors

itsNatTan committed:
Add GUI

WZWren commented on src/main/java/task/Deadline.java:
```diff
@@ -0,0 +1,148 @@
+    /**
+     * @return A boolean on whether deadline is overdue or not
+     */
+    public boolean overdue() {
```
Perhaps you should change this method name to be more boolean-like, like `isOverdue`?

WZWren commented on src/main/java/task/Deadline.java:
```diff
@@ -0,0 +1,148 @@
+     * @param localDateTime Date and Time of the deadline
+     * @return boolean
+     */
+    public boolean dueBy(LocalDateTime localDateTime) {
```
Similar issue to overdue, this time it could be something like `isDueBy`? That way it would feel more like a question, ie is `[task name]` due by `[localDateTime param]`?

WZWren commented on src/main/java/util/Parser.java:
```diff
@@ -0,0 +1,172 @@
+     *
+     * @param command User Input
+     */
+    public String toDoParser(String command) {
```
Maybe make the toDo a composite word, rather than 2 distinct words for camelCase? When I first saw the method, I thought it would convert this to a DoParser object of some kind, which confused me a lot.

WZWren replied:
Separately, wouldn't this be more intuitive to name as parseTodo? Right now it sounds like an object that I am fetching, rather than a method that parses Strings to Todos. This applies for the rest of the task parsers.

itsNatTan committed:
Add assertions for important assumptions

There are several methods that Parser will call based on the command word.
By using assertions, I can ensure that Parser will route to the right command.

itsNatTan committed:
Made code quality improvements based on PR review

itsNatTan committed:
Merge pull request #4 from itsNatTan/branch-A-Assertions

Add assertions for important assumptions

itsNatTan committed:
Merge branch 'master' into branch-A-CodeQuality

itsNatTan committed:
Merge pull request #5 from itsNatTan/branch-A-CodeQuality

Made code quality improvements based on PR review

itsNatTan committed:
Made code quality improvements based on PR review

itsNatTan committed:
Merge pull request #6 from itsNatTan/branch-C-DetectDuplicates

Add function to detect duplicates and prevent addition to tasklist

itsNatTan committed:
Create Fat JAR file
Add load method for Storage to load from .txt file

itsNatTan committed:
Fix bug when overdue or dueby shows completed tasks

itsNatTan committed:
Add Ui.png as representative screenshot

itsNatTan committed:
Update README.md

itsNatTan committed:
Reduce nesting at dueby function using guard clauses

itsNatTan committed:
Merge remote-tracking branch 'origin/master'

itsNatTan committed:
Improve GUI to expand dialogbox when content is too much
Improve GUI to close window when "bye" is the user input

itsNatTan committed:
Fix format of greeting text

itsNatTan committed:
Update README.md
